frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInBranchBelowAction.java

```diff
@@ -230,27 +157,62 @@ String createOrCheckoutNewBranch(Project project, GitRepository gitRepository, S
         initialName,
         /* showCheckOutOption */ true,
         /* showResetOption */ true,
-        /* showSetTrackingOption */ true);
+        /* showSetTrackingOption */ false); // TODO (#496): use setTrackingOption that comes with 2020.2
 
     GitNewBranchOptions options = gitNewBranchDialog.showAndGetOptions();
 
     if (options == null) {
       log().debug("Name of branch to slide in is null: " +
           "most likely the action has been canceled from create-new-branch dialog");
-      return null;
+      return Tuple.of(null, () -> {});
     }
 
     var branchName = options.getName();
     if (!initialName.equals(branchName)) {
-      return branchName;
+      return Tuple.of(branchName, () -> {});
     }
 
-    if (options.shouldCheckout()) {
-      checkoutOrReset(project, repositories, startPoint, options);
-    } else {
-      createNewBranch(project, repositories, startPoint, options);
+    Runnable preSlideInRunnable = () -> {};
+    var remoteBranch = getGitRemoteBranch(project, gitRepository, branchName);
+
+    if (options.shouldCheckout() && remoteBranch != null) {
+      preSlideInRunnable = () -> checkoutRemoteBranch(project, repositories, remoteBranch.getName());
+
+    } else if (!options.shouldCheckout() && remoteBranch != null) {
+      var refspec = "refs/remotes/${remoteBranch.getName()}:refs/heads/${branchName}";
+      preSlideInRunnable = () -> new FetchBackgroundable(project, gitRepository, LOCAL_REPOSITORY_NAME, refspec,
+          "Fetching Remote Branch").queue();
+
+    } else if (options.shouldCheckout() && remoteBranch == null) {
+      preSlideInRunnable = () -> checkoutOrReset(project, repositories, startPoint, options);
+
+    } else if (!options.shouldCheckout() && remoteBranch == null) {
+      preSlideInRunnable = () -> createNewBranch(project, repositories, startPoint, options);
     }
 
-    return options.getName();
+    return Tuple.of(options.getName(), preSlideInRunnable);
+  }
+
+  @Nullable
+  private static GitRemoteBranch getGitRemoteBranch(Project project, GitRepository gitRepository, String branchName) {
+    var remotes = gitRepository.getRemotes();
+    if (remotes.isEmpty()) {
+      return null;
+    }
+
+    var remotesWithBranch = List.ofAll(remotes).flatMap(r -> {
+      var remoteBranchName = "${r.getName()}/${branchName}";
+      var remoteBranch = gitRepository.getBranches().findRemoteBranch(remoteBranchName);
+      return remoteBranch != null ? Option.some(Tuple.of(r, remoteBranch)) : Option.none();
+    }).sortBy(t -> !t._1().getName().equals("origin"));
```

Comment:
Hmm is `false < true` or `true < false`?...

Reply:
Pls drop a comment

Reply:
Ok, I see now it must be the former... anyway, that's a bit non-trivial at first glance